internal/querynode/data_sync_service.go

```diff
@@ -297,35 +184,30 @@ func newDataSyncService(ctx context.Context,
 	factory msgstream.Factory) *dataSyncService {
 
 	return &dataSyncService{
-		ctx:                       ctx,
-		collectionFlowGraphs:      make(map[UniqueID]map[Channel]*queryNodeFlowGraph),
-		collectionDeltaFlowGraphs: map[int64]map[string]*queryNodeFlowGraph{},
-		partitionFlowGraphs:       make(map[UniqueID]map[Channel]*queryNodeFlowGraph),
-		streamingReplica:          streamingReplica,
-		historicalReplica:         historicalReplica,
-		tSafeReplica:              tSafeReplica,
-		msFactory:                 factory,
+		ctx:               ctx,
+		dmlFlowGraphs:     make(map[Channel]*queryNodeFlowGraph),
+		deltaFlowGraphs:   make(map[Channel]*queryNodeFlowGraph),
+		streamingReplica:  streamingReplica,
+		historicalReplica: historicalReplica,
+		tSafeReplica:      tSafeReplica,
+		msFactory:         factory,
 	}
 }
 
 // close would close and remove all flow graphs in dataSyncService
 func (dsService *dataSyncService) close() {
-	// close collection flow graphs
-	for _, nodeFGs := range dsService.collectionFlowGraphs {
-		for _, nodeFG := range nodeFGs {
-			if nodeFG != nil {
-				nodeFG.flowGraph.Close()
-			}
+	// close DML flow graphs
+	for _, nodeFG := range dsService.dmlFlowGraphs {
+		if nodeFG != nil {
+			nodeFG.flowGraph.Close()
 		}
 	}
-	// close partition flow graphs
-	for _, nodeFGs := range dsService.partitionFlowGraphs {
-		for _, nodeFG := range nodeFGs {
-			if nodeFG != nil {
-				nodeFG.flowGraph.Close()
-			}
+	dsService.dmlFlowGraphs = make(map[Channel]*queryNodeFlowGraph)
+	// close delta flow graphs
+	for _, nodeFG := range dsService.deltaFlowGraphs {
+		if nodeFG != nil {
+			nodeFG.flowGraph.Close()
 		}
 	}
-	dsService.collectionFlowGraphs = make(map[UniqueID]map[Channel]*queryNodeFlowGraph)
-	dsService.partitionFlowGraphs = make(map[UniqueID]map[Channel]*queryNodeFlowGraph)
+	dsService.deltaFlowGraphs = make(map[Channel]*queryNodeFlowGraph)
```

Comment:
It's confused to map a new map when close dataSyncService, why not delete dsService.deltaFlowGraphs by channel after flowGraph.Close()?

Reply:
make a new map then the original map will be garbage-collected.
see <https://newbedev.com/how-to-clear-a-map-in-go>